File: src/main/java/com/project/pantau/dto/report/UpdateReportRequest.java
package com.project.pantau.dto.report;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import org.springframework.web.multipart.MultipartFile;

public record UpdateReportRequest(
        @NotNull(message = "Category id cannot be null")
        Long categoryId,

        @Size(max = 2000, message = "Maximum report description is 2000 characters")
        String description,

        MultipartFile photo,

        @NotNull(message = "Latitude cannot be null")
        Double latitude,

        @NotNull(message = "Longitude cannot be null")
        Double longitude
) {
}

File: backend/internal/dto/report/update_report_request.go
package report

import "mime/multipart"

type UpdateReportRequest struct {
	CategoryID  *int64                  `json:"category_id" form:"category_id" validate:"required"`
	Description *string                 `json:"description" form:"description" validate:"omitempty,max=2000"`
	Photos      []*multipart.FileHeader `json:"photos" form:"photos" validate:"omitempty,max=4"`
	Latitude    *float64                `json:"latitude" form:"latitude" validate:"required"`
	Longitude   *float64                `json:"longitude" form:"longitude" validate:"required"`
}
